lib/spack/spack/build_environment.py

```diff
@@ -104,6 +105,17 @@
 dso_suffix = 'dylib' if sys.platform == 'darwin' else 'so'
 
 
+@contextmanager
+def swap_to_frontend(frontend, backend):
```

Comment:
Can we name this something more descriptive?  I think `swap_to_frontend` is a little confusing and spack-specific, in that it requires people to know a lot about the build model to understand what it really does.

What about `compile_for_build_time()` or `compile_build_tools()` or `build_environment_compilers()`?

Reply:
Another thing about this context manager is that it requires that you tell it both the `frontend` and the `backend`.  I suspect that some packages are going to want to use this as well, but they shouldn't have to know the frontend and the backend.  Ideally a package writer would write:

```python
with build_env_compilers():
    # build various build tools so that they'll run during the build
    # if we build them with cross compilers, they'll fail

# now, outside the build environment, run make with CC and friends set to the cross-compiler
make()
```

I realize the use case above is *slightly* different from what Cray recommends, but it's one that comes up for other projects, and the solution is similar.